api/client/webclient/webclient_test.go

```diff
@@ -442,3 +444,57 @@ func TestSSHProxyHostPort(t *testing.T) {
 		})
 	}
 }
+
+// TestWebClientClosesIdleConnections verifies that all http connections
+// are closed when the http.Client created by newWebClient is no longer
+// being used.
+func TestWebClientClosesIdleConnections(t *testing.T) {
+	expectedResponse := &PingResponse{
+		Proxy: ProxySettings{
+			TLSRoutingEnabled: true,
+		},
+		ServerVersion:    "1.2.3",
+		MinClientVersion: "0.1.2",
+		ClusterName:      "test",
+	}
+
+	expectedStates := []http.ConnState{
+		http.StateNew, http.StateActive, http.StateClosed, // the https request will fail and cause us to fallback to http
+		http.StateNew, http.StateActive, http.StateIdle, http.StateClosed, // the http request should be processed and closed
+	}
+
+	srv := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/webapi/find":
+			json.NewEncoder(w).Encode(expectedResponse)
+		default:
+			w.WriteHeader(http.StatusBadRequest)
+		}
+	}))
+
+	stateChange := make(chan http.ConnState, len(expectedStates))
+	srv.Config.ConnState = func(conn net.Conn, state http.ConnState) {
+		stateChange <- state
+	}
+
+	srv.Start()
+	t.Cleanup(srv.Close)
+
+	resp, err := Find(&Config{
+		Context:   context.Background(),
+		ProxyAddr: srv.URL[len("http://"):],
+		Insecure:  true,
+	})
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(expectedResponse, resp))
+
+	for _, expected := range expectedStates {
+		select {
+		case state := <-stateChange:
+			require.Equal(t, expected, state, "expected connection state %s got %s", expected.String(), state.String())
+		case <-time.After(10 * time.Second):
+			t.Fatalf("timeout waiting for expected connection state %s", expected.String())
+		}
```

Comment:
This test runs really fast (0.007s)
:nail_care: Can we decrease this timeout to something like 1s? Just to increase feedback time if something changes and it starts to fail.